pkg/gather/gather.go

```diff
@@ -44,8 +44,9 @@ type GathererFunctionReport struct {
 
 // ArchiveMetadata contains the information about the archive and all its' gatherers
 type ArchiveMetadata struct {
-	// info about gathering functions
-	StatusReports []GathererFunctionReport `json:"status_reports"`
+	// info about gathering functions. We need to use map, because functions will repeat
```

Comment:
We actually don't have to, it could stay an array, but internally we can use a map so that we use only the latest status. Btw, the motivation behind storing all the results was that they can be different(for whatever reason) and we would store all of them.

Reply:
Well I think it's very confusing to store results of failed gatherer which result is not recorded (in the archive), but maybe it's a topic for discussion.
OK for using the map only internally.